pom.xml

```
@@ -1,356 +1,687 @@
-            <url>https://raw.githubusercontent.com/NationalSecurityAgency/datawave/mvn-repo</url>
-        </repository>
-    </repositories>
+      <releases>
+        <enabled>true</enabled>
+      </releases>
+      <snapshots>
+        <enabled>false</enabled>
+      </snapshots>
+      <id>datawave-github-mvn-repo</id>
+      <url>https://raw.githubusercontent.com/NationalSecurityAgency/datawave/mvn-repo</url>
+    </repository>
+  </repositories>
+  <build>
+    <pluginManagement>
+      <plugins>
+        <plugin>
+          <groupId>com.github.ekryd.sortpom</groupId>
+          <artifactId>sortpom-maven-plugin</artifactId>
+          <version>2.15.0</version>
+        </plugin>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-enforcer-plugin</artifactId>
+          <version>3.2.1</version>
+          <executions>
+            <execution>
+              <id>check-maven-version</id>
+              <goals>
+                <goal>enforce</goal>
+              </goals>
+              <configuration>
+                <rules>
+                  <requireMavenVersion>
+                    <version>[${maven.min-version},)</version>
+                  </requireMavenVersion>
+                  <requireJavaVersion>
+                    <version>[1.8,)</version>
+                  </requireJavaVersion>
+                  <dependencyConvergence />
+                </rules>
+              </configuration>
+            </execution>
+          </executions>
+        </plugin>
+      </plugins>
+    </pluginManagement>
+    <plugins>
+      <plugin>
+        <!-- verify before compile; should be sorted already -->
+        <groupId>com.github.ekryd.sortpom</groupId>
+        <artifactId>sortpom-maven-plugin</artifactId>
+        <configuration>
+          <predefinedSortOrder>recommended_2008_06</predefinedSortOrder>
+          <createBackupFile>false</createBackupFile>
+          <lineSeparator>\n</lineSeparator>
+          <expandEmptyElements>false</expandEmptyElements>
+          <nrOfIndentSpace>2</nrOfIndentSpace>
+          <sortDependencies>scope,groupId,artifactId</sortDependencies>
+          <sortDependencyExclusions>scope,groupId,artifactId</sortDependencyExclusions>
+          <sortPlugins>groupId,artifactId</sortPlugins>
+          <sortProperties>true</sortProperties>
+        </configuration>
+        <executions>
+          <execution>
+            <id>sort-pom</id>
+            <goals>
+              <goal>sort</goal>
+            </goals>
+            <phase>process-sources</phase>
+          </execution>
+          <execution>
+            <id>verify-sorted-pom</id>
+            <goals>
+              <goal>verify</goal>
+            </goals>
+            <phase>process-resources</phase>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-enforcer-plugin</artifactId>
```

```suggestion
        <artifactId>maven-enforcer-plugin</artifactId>
        <version>3.2.1</version>
```
This seems to address both forcing this project to use 3.2.1 vs 3.0.0-M3 from the parent and using 3.2.1 seems to also solve the problem with 3.5.4 maven compatibility I was seeing.

---

We may need to keep in mind that all of the enforcer goals are being leveraged in this project:

from the microservice-parent: https://github.com/NationalSecurityAgency/datawave-parent/blob/df56fa23e92835a3e8e790aa3584016440398e56/pom.xml#L322
from the microservice-service-parent: https://github.com/NationalSecurityAgency/datawave-service-parent/blob/main/pom.xml#L147

There might be a future issue with both the microservice parent `enforce-no-snapshots` and this parent `enforce-maven-version` enforcing java versions if the top level parent moves ahead of this project parent.